### DISCUSSION TITLE
Kratos selfservice ui with hydra and react native

### DISCUSSION THREAD

**Discussion by Dakraeth2** [ory/kratos#1200] [Q&A] [answered]
Hi I currently struggle to get  Kratos selfservice ui with hydra-integration-2021 from https://github.com/ory/kratos-selfservice-ui-node/tree/hydra-integration-2021 to work with react native.
I've cloned the files from https://github.com/FormidableLabs/react-native-app-auth and added following configuration.
```javascript
hydra: {
    issuer: 'http://my-hydra-ip:4444',
    clientId: 'interactive.public',
    redirectUrl: 'io.identityserver.demo:/oauthredirect',
    additionalParameters: {},
    scopes: ['openid', 'profile'],
    dangerouslyAllowInsecureHttpRequests: true
  }
```
  and this button
```javascript
<Button
    onPress={() => handleAuthorize('hydra')}
    text="Authorize Hydra"
    color="#DA2536"
/>
```
            
 I've created the hydra client via
```bash
sudo docker-compose exec hydra \
    hydra clients create \
    --endpoint http://my-hydra-ip:4445 \
    --id interactive.public \
    --grant-types authorization_code \
    --response-types code,id_token \
    --scope openid,offline,profile \
    --token-endpoint-auth-method none \
    --callbacks io.identityserver.demo:/oauthredirect
```
    
  When I press the Button the Kratos selfservice ui opens and I can enter my credentials. When submitting i got redirected to https://www.ory.sh/kratos/docs/fallback/error/

Isnt it possible to redirect to a react-native schema based url?

Or am I missing any significant?
Thanks

**Comment by aeneasr** [comment, marked as answer]
Could you share your kratos logs please?

**Reply by Dakraeth2** [reply]
This are the logs from the time I pressed login button

[kratos_logs.txt](https://github.com/ory/kratos/files/6254359/kratos_logs.txt)

**Reply by aeneasr** [reply]
looks like you are encountering a CDRF issue

**Reply by Dakraeth2** [reply]
Thanks for the hint. What would be a good way to test?
I've only cloned the two repos and changed the IP-Addresses to my public server ones.

**Reply by Dakraeth2** [reply]
What I've noticed is on my desktop browser the self-service-ui works very vell, but on my mobile phone after submitting the credentials i got redirected to this error page

**Reply by aeneasr** [reply]
without details impossible to say

**Reply by Dakraeth2** [reply]
what details needed?

**Reply by aeneasr** [reply]
A "mobile phone" is very generic. What mobile phone, what browser, what logs, what flows? Anything you would need yourself to debug this stuff.

**Reply by Dakraeth2** [reply]
Okay I've tested a bit more and figured out it wasn't the phone itself. The thing is when I use Firefox on my mobile or deskop environment it works. But when I use Chrome, Opera or Edge I got redirected to the error page.
So the error does even happen when I try login in to the self-service-ui without react-native. I think I can exclude the react native  as failing part.

Even if i think it isnt neccesarry anymore the phone is an One Plus Nord with Android 11.

**Reply by aeneasr** [reply]
Ok, I recommend checking out https://www.ory.sh/kratos/docs/debug/csrf#common-pitfalls

**Reply by aeneasr** [reply]
And: https://www.ory.sh/kratos/docs/next/debug/csrf#samesite-attribute

**Comment by vinckr** [comment]
Hey @Dakraeth2 
The Ory Kratos and Ory Hydra integration just landed with Hydra 2.0, 
see the [release notes](https://github.com/ory/hydra/releases/tag/v2.0.0) for more details!
